cpplint.py

```diff
@@ -50,12 +50,23 @@
 import sre_compile
 import string
 import sys
+import itertools
 import unicodedata
 
+# Files with any of these extensions are considered to be
+# header files (and will undergo different style checks).
+# This set can be extended by using the --headers
+# option (also supported in CPPLINT.cfg)
+_header_extensions = set(['h', 'hpp', 'hxx', 'h++', 'cuh'])
+_nonheader_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'cu'])
+
+
 # The allowed extensions for file names
 # This is set by --extensions flag.
-_valid_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'h', 'hpp', 'hxx',
-    'h++'])
+_valid_extensions = _nonheader_extensions.union(_header_extensions)
```

Comment:
what's troubling me now is that this variable will be overridden by `--extensions` (or the CPPLINT.cfg `extensions=...`), breaking the relation between it and the `header_extensions` and `nonheader_extensions` variables. this will cause some unexpected behaviors...

I suggest that instead you enforce consistency, by not allowing this variable to be overridden directly, but by replacing it with flags and CPPLINT.cfg options for the separate sets (e.g. `--headers` and `--nonheaders`), and making sure to update `_valid_extensions` accordingly when modifying these.